sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -96,8 +96,10 @@ class SparkSessionExtensions {
   type ParserBuilder = (SparkSession, ParserInterface) => ParserInterface
   type FunctionDescription = (FunctionIdentifier, ExpressionInfo, FunctionBuilder)
   type ColumnarRuleBuilder = SparkSession => ColumnarRule
+  type QueryStagePrepRuleBuilder = SparkSession => Rule[SparkPlan]
```

Comment:
Please update the document (at line 47) accordingly.

Reply:
Thanks. I pushed an update for this.

Reply:
This is a public API. I think we also need to add a version information.